lib/core/elastic_connector_actions.rb

```diff
@@ -72,10 +85,30 @@ def set_configurable_field(connector_id, field_name, label, value)
       end
 
       def claim_job(connector_id)
-        update_connector_fields(connector_id,
-                                :sync_now => false,
-                                :last_sync_status => Connectors::SyncStatus::IN_PROGRESS,
-                                :last_synced => Time.now)
+        seq_no = nil
+        primary_term = nil
+        sync_in_progress = false
+        client.get(
+          :index => Utility::Constants::CONNECTORS_INDEX,
+          :id => connector_id,
+          :ignore => 404,
+          :refresh => true
+        ).tap do |response|
+          seq_no = response['_seq_no']
+          primary_term = response['_primary_term']
+          sync_in_progress = response.dig('_source', 'last_sync_status') == Connectors::SyncStatus::IN_PROGRESS
+        end
+        if sync_in_progress
+          raise JobAlreadyRunningError.new(connector_id)
+        end
```

Comment:
I think we should create a bug for this case.

Reply:
I think @wangch079 is addressing these types of scenarios in his current project. I wouldn't worry too much about trying to make an issue or fix it as a one off, since it'll be covered holistically.

Reply:
> I think we should create a bug for this case.

@mchernyavskaya  Is there a followup bug for this?

Reply:
@danajuratoni based on @seanstory 's comment, I didn't think that I should create one.

Reply:
@wangch079 actually, a question for you. Can your current work help prevent the situation when the connector that's died or killed forcefully mid-sync will never be able to sync again?